### PR TITLE
JAMES-2937 Dequeue should work after network outage 

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java [hunk 1]
```diff
@@ -106,11 +119,28 @@ public ReactorRabbitMQChannelPool(Mono<Connection> connectionMono, int poolSize)
...
+    public Mono<Connection> getConnectionMono() {
+        return connectionMono;
+    }
+
     @Override
     public Mono<? extends Channel> getChannelMono() {
         return Mono.fromCallable(() -> {
             Channel channel = pool.borrowObject(MAXIMUM_BORROW_TIMEOUT_IN_MS);
             borrowedChannels.add(channel);
+            Preconditions.checkArgument(channel.isOpen());
```
Comment (blackheaven) [inline backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java]: Maybe one line up

File: mailbox/event/event-rabbitmq/src/test/java/org/apache/james/mailbox/events/RabbitMQEventBusTest.java [hunk 2]
```diff
@@ -129,17 +119,16 @@ void tearDown() {
         eventBus3.stop();
         ALL_GROUPS.stream()
             .map(GroupRegistration.WorkQueueName::of)
-            .forEach(queueName -> sender.delete(QueueSpecification.queue(queueName.asString())).block());
-        sender.delete(ExchangeSpecification.exchange(MAILBOX_EVENT_EXCHANGE_NAME)).block();
-        sender.close();
+            .forEach(queueName -> rabbitMQExtension.getRabbitChannelPool().getSender().delete(QueueSpecification.queue(queueName.asString())).block());
+        rabbitMQExtension.getRabbitChannelPool().getSender().delete(ExchangeSpecification.exchange(MAILBOX_EVENT_EXCHANGE_NAME)).block();
```
Comment (blackheaven) [inline mailbox/event/event-rabbitmq/src/test/java/org/apache/james/mailbox/events/RabbitMQEventBusTest.java]: CRs please

File: server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueFactory.java [hunk 3]
```diff
@@ -159,7 +170,22 @@ public RabbitMQMailQueue createQueue(String name) {
...
+        Flux.concat(
+            reactorRabbitMQChannelPool.getSender().declareExchange(ExchangeSpecification.exchange(exchangeName)
+                .durable(true)
+                .type("direct")),
+            reactorRabbitMQChannelPool.getSender().declareQueue(QueueSpecification.queue(mailQueueName.toWorkQueueName().asString())
+                .durable(DURABLE)
+                .exclusive(!EXCLUSIVE)
+                .autoDelete(!AUTO_DELETE)
+                .arguments(NO_ARGUMENTS)),
+            reactorRabbitMQChannelPool.getSender().bind(BindingSpecification.binding()
```
Comment (blackheaven) [inline server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueFactory.java]: CR

**Comment by chibenwa** [comment]
test this please

(build output lost)

File: backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java (hunk 1 above)
Comment (chibenwa) [reply backends-common/rabbitmq/src/main/java/org/apache/james/backends/rabbitmq/ReactorRabbitMQChannelPool.java]: Should be on https://github.com/linagora/james-project/pull/2741

File: mailbox/event/event-rabbitmq/src/test/java/org/apache/james/mailbox/events/RabbitMQEventBusTest.java (hunk 2 above)
Comment (chibenwa) [reply mailbox/event/event-rabbitmq/src/test/java/org/apache/james/mailbox/events/RabbitMQEventBusTest.java]: Should be on https://github.com/linagora/james-project/pull/2741

File: server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueFactory.java (hunk 3 above)
Comment (chibenwa) [reply server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueFactory.java]: Should be on #2741

**Comment by Arsnael** [comment]
Read it

**Comment by Arsnael** [comment]
CI restart...

test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**